nats-base-client/util.ts

```diff
@@ -200,3 +202,31 @@ export function shuffle(a: any[]): any[] {
   }
   return a;
 }
+
+export function encodeHeader(headers: Headers): Uint8Array {
+  let v = `NATS/1.0\r\n`;
+  for (const [key, value] of headers) {
+    v += `${key}: ${value}\r\n`;
```

Comment:
See comment about decode. You may have to repeat the same key with a different value.
Also, the server generally does not check headers content, but we already have a case where it will check for a Msg-Id header for JetStream de-dup feature. Since it does a byte scan of this header, it will not be tolerant of the casing which it assumes will be canonical ("my-key" becomes "My-Key"). Libraries are therefore expected to send the headers in a canonical form. (see thread here: https://github.com/nats-io/nats-server/pull/1508#discussion_r452382516)